events.go

```diff
@@ -286,26 +286,26 @@ type EventDeploymentFailed struct {
 
 // EventDeploymentInfo describes a 'deployment_info' event.
 type EventDeploymentInfo struct {
-	EventType   string          `json:"eventType"`
-	CurrentStep *DeploymentStep `json:"currentStep"`
-	Timestamp   string          `json:"timestamp"`
-	Plan        *DeploymentPlan `json:"plan"`
+	EventType   string            `json:"eventType"`
+	CurrentStep *DeploymentStepV1 `json:"currentStep"`
```

Comment:
I was about to ask you why you'd find a rename to `CurrentStep` confusing when I realized that for the event types, `currentStep` (the JSON property) is the kind of object we've been talking about mostly whereas in the `deployments` endpoint response, the same property denotes an integer. That's an unfortunate choice on Marathon's end in my opinion.

Regarding option 1: As you suspected, I'd rather not keep using the `V1` suffix as we figured out that `DeploymentStepV1` (being used for events already) is not specific to Marathon 1.x at all, so it'd give a wrong impression and likely lead to confusion. We'd probably want to rename it again at some point in the future, which is going to be another (albeit small) breaking change.

Regarding option 3: While I like this approach in general, AFAICS it means that we'd need to change the type that `Deployment.Steps` points to. That's something we deliberately avoided during #158 because we were not sure if the breaking API change described in [upstream issue 3965](github.com/mesosphere/marathon/issues/3965) would be reverted or not. The labelling on that issue seems to indicate that things will be reverted; I [asked for clarification again earlier today](https://github.com/mesosphere/marathon/issues/3965#issuecomment-236159923) and hope to get some feedback. Until then, I'd like to maintain the externally visible structure of the `Deployment` struct and not suffix a type with `V0` when there's still a possibility it will eventually become the canonical type for Marathon version 1.

Regarding option 2: I believe that `EventDeployment.*.CurrentStep` and `Deployment.Steps` are supposed to reference the same type, so duplicating `DeploymentStepV1` under a different name and decoupling the two may lead to additional maintenance effort. My suggestion would be to rename the struct `DeploymentStepV1` to something like `DeploymentStepActions` or `StepActions` (i.e., describe what it actually is instead of trying to match its corresponding field name) and use that for the events and in [marathon.Deployments()](https://github.com/gambol99/go-marathon/blob/2e73c1d824745b3b6374f2d7cc46932f5a596698/deployment.go#L91). Since `DeploymentStepV1` is only used internally, we wouldn't break anyone now nor later if upstream decides to fix the breaking change by means of reverting.

So my opinion would be to go with option 2 but rename instead of clone/duplicate.

What do you think?